mapbox/android-docs
# [tests] add copyeditor

Ref https://github.com/mapbox/docs-test-suite/issues/13

This PR adds the [@mapbox/copyeditor](https://github.com/mapbox/copyeditor) to assert the documentation styleguide and check spelling.

This PR also adds:
* `*.dic` -  to store words for this repo's personal dictionary.
* `.copyeditorrc` - copyeditor's config file. Here we can enable the optional plugins (retext-spell, -equality, -simplify, -passive) and list any rules we want to allow that may have failed on any of the plugins. We also define the path to this repo's personal dictionary.

~~💡 This PR does not enable retext-passive, I'll create a separate PR for those fixes.~~

How to test:
1. Pull down this branch.
2. `npm install`
3. `npm test` or `npm run test-content`

Questions for the reviewer:
* **Spellcheck:**
   - [ ] Are there any words in `.dic` that are incorrect and actually misspelled?
   - [x] Should we add any of these words to copyeditor's global dictionary?
* **Allowed rules**: In `.copyeditorrc`, we can list rules that we think should be allowed given the context they are used in.
   - [x] Are the rules in `allow` permissible in this repo or should we find alternatives? You can test by deleting a word from the `allow` array and then running `npm test` to see where the rule fails.
   - [x] Should we add any of the allowed rules in the global copyeditor's global config?
* [ ] Do the copyedits read ok?
